src/sync.rs

```diff
@@ -17,6 +17,8 @@ pub(crate) fn fence(ord: Ordering) {
 }
 
 #[cfg(not(loom))]
-pub(crate) use std::sync::atomic::{fence, AtomicPtr, AtomicUsize, Ordering};
+pub(crate) use alloc::sync::Arc;
 #[cfg(not(loom))]
-pub(crate) use std::sync::{Arc, Mutex, MutexGuard};
+pub(crate) use core::sync::atomic::{fence, AtomicPtr, AtomicUsize, Ordering};
+#[cfg(not(loom))]
+pub(crate) use std::sync::{Mutex, MutexGuard};
```

Comment:
Just a note to self that this is still a `std` dependency as we've discussed in #101.